src/runtime.cc

```diff
@@ -60,16 +62,16 @@ namespace detail {
 #endif
 	}
 
-	std::string get_sycl_version() {
-		std::stringstream ss;
+	const char* get_sycl_version() {
 #if defined(__COMPUTECPP__)
-		ss << "ComputeCpp " << COMPUTECPP_VERSION_MAJOR << "." << COMPUTECPP_VERSION_MINOR << "." << COMPUTECPP_VERSION_PATCH;
-		return ss.str();
+		return "ComputeCpp " CELERITY_STRINGIFY(COMPUTECPP_VERSION_MAJOR) "." CELERITY_STRINGIFY(COMPUTECPP_VERSION_MINOR) //
+		    "." CELERITY_STRINGIFY(COMPUTECPP_VERSION_PATCH);
 #elif defined(__HIPSYCL__) || defined(__HIPSYCL_TRANSFORM__)
-		ss << "hipSYCL " << HIPSYCL_VERSION_MAJOR << "." << HIPSYCL_VERSION_MINOR << "." << HIPSYCL_VERSION_PATCH;
-		return ss.str();
+		return "hipSYCL " CELERITY_STRINGIFY(HIPSYCL_VERSION_MAJOR) "." CELERITY_STRINGIFY(HIPSYCL_VERSION_MINOR) "." CELERITY_STRINGIFY(HIPSYCL_VERSION_PATCH);
+#elif CELERITY_DPCPP
+		return "DPC++ / Clang " __clang_version__;
 #else
-		return "Unknown";
+#error "unknown SYCL implementation"
```

Comment:
I think this is a noteworthy behavior change. This commit will prevent Celerity from compiling on unknown SYCL platforms, while previously it would.

Do we want this?

Reply:
Yes we discussed this offline; the reason why this bug went unnoticed is because we fell back to `"Unknown"` here. This was an idealistic decision made back in the day, when in reality you cannot compile Celerity against a new SYCL implementation without explicitly supporting it in our CMake infrastructure first.

Reply:
Yeah, I agree with this point. I wanted to note it down though since it's a somewhat significant change that might not be implied by the commit/pull request names.